nwb_conversion_tools/datainterfaces/axonadatainterface.py

```diff
@@ -0,0 +1,316 @@
+"""Authors: Steffen Buergers"""
+import os
+import re
+import datetime
+import contextlib
+import mmap
+import numpy as np
+
+import spikeextractors as se
+
+from pynwb import NWBFile
+from pynwb.behavior import Position, SpatialSeries
+
+from ..basedatainterface import BaseDataInterface
+from ..baserecordingextractorinterface import (
+    BaseRecordingExtractorInterface
+)
+
+from spikeextractors.extractors.nwbextractors.nwbextractors import check_module
+
+
+# Helper functions for AxonaRecordingExtractorInterface
+def parse_generic_header(filename, params):
+    """
+    Given a binary file with phrases and line breaks, enters the
+    first word of a phrase as dictionary key and the following
+    string (without linebreaks) as value. Returns the dictionary.
+
+    INPUT
+    filename (str): .set file path and name.
+    params (list or set): parameter names to search for.
+
+    OUTPUT
+    header (dict): dictionary with keys being the parameters that
+                   were found & values being strings of the data.
+
+    EXAMPLE
+    parse_generic_header('myset_file.set', ['experimenter', 'trial_time'])
+    """
+    header = {}
+    params = set(params)
+    with open(filename, 'rb') as f:
+        for bin_line in f:
+            if b'data_start' in bin_line:
+                break
+            line = bin_line.decode('cp1252').replace('\r\n', '').\
+                replace('\r', '').strip()
+            parts = line.split(' ')
+            key = parts[0]
+            if key in params:
+                header[key] = ' '.join(parts[1:])
+
+    return header
+
+
+def read_iso_datetime(set_file):
```

Comment:
```suggestion
def read_axona_iso_datetime(set_file):
```
Very minor, but a name like `read_iso_datetime` implies it has a general purpose of converting any arbitrary ISO into a datetime, whereas this seems very particular to axona. Which is why it's in the axona interface itself, and not the general utils, but still, uniquely named functions avoid future confusion.